jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/HttpInput.java

```diff
@@ -257,6 +277,16 @@ public int read() throws IOException
 
     @Override
     public int read(byte[] b, int off, int len) throws IOException
+    {
+        return read(null, b, off, len);
+    }
+
+    public int read(ByteBuffer buffer) throws IOException
+    {
+        return read(buffer, null, -1, -1);
+    }
+
+    private int read(ByteBuffer buffer, byte[] b, int off, int len) throws IOException
```

Comment:
Weird signature.

What happens if it is submitted with a `buffer` not null, but a `off` == 10 and `len` == 600 provided?